### PR TITLE
Test scenario service

### PR DESCRIPTION
**Pull request by IfLunatic**
Added tests, changed some imports to be more correct, and added an annotation

### CODE REVIEW

File: src/test/java/com/geeksforless/scenario/ScenarioServiceImplTest.java [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    void testSendScenario() {
+        ScenarioFormDTO formData = new ScenarioFormDTO("[]", false);
+        ScenarioRequestDTO expectedRequest = ScenarioRequestDTO.builder()
+                .userId(1L)
+                .scenarioList(Collections.emptyList())
+                .proxyRequired(false)
+                .build();
+        scenarioService.sendScenario(formData, authentication);
+        verify(workerClient).postScenarios(expectedRequest);
+        verifyNoMoreInteractions(workerClient);
```
Comment (jurok3x) [inline src/test/java/com/geeksforless/scenario/ScenarioServiceImplTest.java:53]: put this in separate method *void tearDown()* with *AfterEach* annotation.

File: src/test/java/com/geeksforless/scenario/ScenarioServiceImplTest.java [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+    @Test
+    void testSendScenario() {
+        ScenarioFormDTO formData = new ScenarioFormDTO("[]", false);
+        ScenarioRequestDTO expectedRequest = ScenarioRequestDTO.builder()
+                .userId(1L)
+                .scenarioList(Collections.emptyList())
+                .proxyRequired(false)
+                .build();
+        scenarioService.sendScenario(formData, authentication);
```
Comment (jurok3x) [inline src/test/java/com/geeksforless/scenario/ScenarioServiceImplTest.java:51]: dont you need to add *doNothing().when(client.postScenarios(any(SendScenarioRequest.class)))*??

File: src/test/java/com/geeksforless/scenario/ScenarioServiceImplTest.java [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+        ScenarioFormDTO formData = new ScenarioFormDTO(validJson, true);
+
+        scenarioService.sendScenario(formData, authentication);
+
+        ScenarioRequestDTO expectedRequest = ScenarioRequestDTO.builder()
+                .userId(1L)
+                .scenarioList(Collections.singletonList(new Scenario("Test scenario", "example.com",
+                        Collections.singletonList(new Step("Click", "Button")))))
+                .proxyRequired(true)
+                .build();
```
Comment (jurok3x) [inline src/test/java/com/geeksforless/scenario/ScenarioServiceImplTest.java:87]: what's difference between this test and the *testSendScenario*?

**Review by jurok3x — CHANGES_REQUESTED** [review]

File: src/test/java/com/geeksforless/scenario/ScenarioServiceImplTest.java (hunk 2 above)
Comment (IfLunatic) [reply src/test/java/com/geeksforless/scenario/ScenarioServiceImplTest.java:51]: The doNothing().when() function is used to specify the behavior of mocks when you need to configure them to do something (for example, throw an exception, return certain values, etc.) when a method is called. But in our case, we only check that the method was called with the expected arguments using verify(workerClient).postScenarios(expectedRequest).

File: src/test/java/com/geeksforless/scenario/ScenarioServiceImplTest.java (hunk 3 above)
Comment (IfLunatic) [reply src/test/java/com/geeksforless/scenario/ScenarioServiceImplTest.java:87]: The main difference is what data is used to create the ScenarioFormDTO and expectedRequest objects, and whether the postScenarios method is called from workerClient depending on the input data.

**Review by jurok3x — APPROVED** [review]

**Pull request merged**